CodingStandard.md

```diff
@@ -1,34 +1,118 @@
+project coding standards, guidelines, and conventions.
+
+If a scenario is not covered on this page, use good judgment and remember *consistency is king*.  If there is an
+established way of doing something in other parts of the code base, that's probably the way it should be done.
+
+If the same question is likely to arise again later in the project, notify the Lead Software Engineer, so he or she can
+update this page for everyone's benefit.
+
+Note that the use of constructs or mechanisms for which justification is required is not discouraged. Requiring 
+justification in these cases is simply intended to prevent misuse and to capture rationale for future reference.
+
+Because the Trickfire Robotics NASA RMC 2018 codebase is built on top of the Robot Operating System framework, any
+topic not covered in this document is deferred to the official ROS C++ style guide, found here:
+<http://wiki.ros.org/CppStyleGuide>
+
+Any topic not covered in either this document or the official ROS style guide is deferred to the official Google C++
+Style Guide, found here: <https://google.github.io/styleguide/cppguide.html>
+
+If a topic is not covered in any of these documents, but it seems that it needs to be addressed, this document may (and
+should) be modified to reflect the most up-to-date standard employed by the Trickfire Robotics Software Team.
+
+General Standards
+=======================================================================================================================
+
+##Naming
+All names should be descriptive of their purpose. Prefer long, descriptive names to short, confusing names.
+
+**Files**
+
+- Source files should be `lower_case`. C++ files should use either the `.h` or `.cpp` file extensions.
+
+- In general, the use of extra prefixes for file names is discouraged:
+  - Source files representing ROS Nodes should not contain a `node_` prefix or `node_` suffix.
+  - Message (`.msg`), Service (`.srv`), and Action (`.action`) files should be `CamelCase`, not contain the respective
+  "Message", "Service", or "Action" identifiers in the name.
+  - Configuration (`.yaml`) files should not contain `_config` or `_param` suffixes.
+  - Launch files (`.launch`) should not contain the `_launch` suffix.
+- Exeptions:
+  - Packages should be `lower_case` and be contain the `tfr_` prefix (as package names are global to the entire ROS
+    ecosystem).
+  - Files defining an abstract base class should contain a `_interface` suffix.
+
+**Identifiers**
+
+C++ code should adhere to the naming conventions below to promote consistency
+throughout the project.
+```
+  - Pre-Processor Macros: `UPPER_CASE`
+  - Class, Struct, Union, and Enum Types: `CamelCase`
+  - Functions and Class Methods: `lowerCamelCase`
+  - Compile-time constants (const, constexpr, enumerators, etc.): `kCamelCase`
+  - Variables and Function Parameters: `lower_case`
+  - Const member variables: `kCamelCase`
```

Comment:
No, I am not a fan of hungarian naming convention we need to get that out of the coding standard.

It hurts readability, it obfuscates intent, it makes code harder to maintain, we need that 100% _out_ of our code base.

Reply:
https://en.wikipedia.org/wiki/Hungarian_notation

Reply:
It makes sense for large mature code bases, but not for this, it sounds like a recipie for a maintainability disaster.